x-pack/plugins/enterprise_search/public/applications/workplace_search/views/groups/groups.tsx

```diff
@@ -71,26 +67,23 @@ export const Groups: React.FC = () => {
   }
 
   const clearFilters = hasFiltersSet && <ClearFiltersLink />;
-  const inviteUsersButton = !isFederatedAuth ? (
+  const inviteUsersButton = (
     <EuiButtonTo to={USERS_PATH} data-test-subj="InviteUsersButton">
       {i18n.translate('xpack.enterpriseSearch.workplaceSearch.groups.inviteUsers.action', {
         defaultMessage: 'Invite users',
       })}
     </EuiButtonTo>
-  ) : null;
-
-  const headerAction = (
-    <EuiFlexGroup responsive={false} gutterSize="m">
-      <EuiFlexItem grow={false}>
-        <EuiButton data-test-subj="AddGroupButton" fill onClick={openNewGroupModal}>
-          {i18n.translate('xpack.enterpriseSearch.workplaceSearch.groups.addGroupForm.action', {
-            defaultMessage: 'Create a group',
-          })}
-        </EuiButton>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>{inviteUsersButton}</EuiFlexItem>
-    </EuiFlexGroup>
   );
+  const createGroupButton = (
+    <EuiButton data-test-subj="AddGroupButton" fill onClick={openNewGroupModal}>
+      {i18n.translate('xpack.enterpriseSearch.workplaceSearch.groups.addGroupForm.action', {
+        defaultMessage: 'Create a group',
+      })}
+    </EuiButton>
+  );
+  const headerActions = !isFederatedAuth
+    ? [inviteUsersButton, createGroupButton]
+    : [createGroupButton];
```

Comment:
Thanks for cleaning this up!

Reply:
While we're here, just wanted to highlight that EUI's `rightSideItems` array is a little unintuitive at first glance - the first item (inviteUsersButton) is actually the rightmost button (matching the old JSX). Anything other items added grow to the left the first item.

Maybe a moot point anyway, IIRC Scotty is removing the `isFederatedAuth` logic soon 🕺